### PR TITLE
VSR: Change cluster id from u32 to u128

### PR DESCRIPTION
**Pull request by sentientwaffle**
Rationale (via https://github.com/tigerbeetle/tigerbeetle/pull/1295#discussion_r1389061252):

> A 128-bit cluster also improves our "resolution" for interoperability with other systems that want to use UUIDs for cluster ID to allow for random generation of cluster ID without the cost of having to introduce a centralized ID coordinator. Plus it gives us a further security guarantee that clusters can't talk to other clusters unless they can literally “guess” the identifier, which is now cryptographically "tough" (granted, we might not be able to imagine now why this could be important, but it's a stronger property nevertheless—especially from an online verification point of view).

### CODE REVIEW

**Comment by sentientwaffle** [comment]
Thanks @batiati for the help on this PR!

File: src/tigerbeetle/main.zig [hunk 1]
```diff
@@ -63,10 +63,6 @@ pub fn main() !void {
     }
 }
 
-// Pad the cluster id number and the replica index with 0s
-const filename_fmt = "cluster_{d:0>10}_replica_{d:0>3}.tigerbeetle";
-const filename_len = fmt.count(filename_fmt, .{ 0, 0 });
```
Comment (sentientwaffle) [inline src/tigerbeetle/main.zig:68]: Dead code!

File: src/clients/java/src/client.zig [hunk 2]
```diff
@@ -194,15 +194,20 @@ comptime {
...
+            assert(env.get_array_length(cluster_id) == 16);
+
+            const cluster_id_elements = env.get_byte_array_elements(cluster_id, null).?;
+            defer env.release_byte_array_elements(cluster_id, cluster_id_elements, .abort);
+
             var context = NativeClient.client_init(
                 false,
                 env,
-                @as(u32, @bitCast(cluster_id)),
+                @as(u128, @bitCast(cluster_id_elements[0..16].*)),
```
Comment (batiati) [inline src/clients/java/src/client.zig:210]: Zig is so beautiful!

File: src/clients/java/src/main/java/com/tigerbeetle/Client.java [hunk 3]
```diff
@@ -25,16 +25,17 @@ public final class Client implements AutoCloseable {
      * @throws InitializationException if an error occurred initializing this client. See
      *         {@link InitializationStatus} for more details.
      *
-     * @throws IllegalArgumentException if {@code clusterID} is negative.
+     * @throws IllegalArgumentException if {@code clusterID} is not a UInt128.
```
Comment (batiati) [inline src/clients/java/src/main/java/com/tigerbeetle/Client.java:29]:
```suggestion
     * @throws NullPointerException if {@code clusterID} is null.
     * @throws IllegalArgumentException if {@code clusterID} is not a UInt128.
```

File: src/clients/java/src/main/java/com/tigerbeetle/Client.java [hunk 4]
```diff
@@ -25,16 +25,17 @@ public final class Client implements AutoCloseable {
...
      *         is null.
      * @throws IllegalArgumentException if {@code concurrencyMax} is zero or negative.
      */
-    public Client(final int clusterID, final String[] replicaAddresses, final int concurrencyMax) {
-        if (clusterID < 0)
-            throw new IllegalArgumentException("ClusterID must be positive");
+    public Client(final byte[] clusterID, final String[] replicaAddresses,
+            final int concurrencyMax) {
+        if (clusterID.length != UInt128.SIZE)
+            throw new IllegalArgumentException("ClusterID must be 16 bytes long");
```
Comment (batiati) [inline src/clients/java/src/main/java/com/tigerbeetle/Client.java:40]:
```suggestion
        Objects.requireNonNull(clusterID, "ClusterID cannot be null");
        if (clusterID.length != UInt128.SIZE)
            throw new IllegalArgumentException("ClusterID must be 16 bytes long");
```

File: src/tigerbeetle/cli.zig [hunk 5]
```diff
@@ -82,7 +82,7 @@ const CliArgs = union(enum) {
         \\        Print this help message and exit.
         \\
         \\  --cluster=<integer>
-        \\        Set the cluster ID to the provided 32-bit unsigned integer.
+        \\        Set the cluster ID to the provided 128-bit unsigned integer.
```
Comment (batiati) [inline src/tigerbeetle/cli.zig]: We should say which format the u128 is expected (eg decimal, hex, guid).

File: src/clients/java/src/main/java/com/tigerbeetle/Client.java (hunk 4 above)
Comment (batiati) [reply src/clients/java/src/main/java/com/tigerbeetle/Client.java:40]: LGTM, just waiting for this one ☝️

Comment (sentientwaffle) [reply src/clients/java/src/main/java/com/tigerbeetle/Client.java:40]: Ah forgot to commit that! Thank you.

**Review by batiati — APPROVED** [review]

**Pull request merged**